### PR TITLE
fix 1 snapshot create a lot of snap file

### PR DESCRIPTION
**Pull request by daresheep**
If the file task more then 1 min.  other request will create new snap file. 

issue:

https://github.com/kubernetes-csi/csi-driver-host-path/issues/243

### CODE REVIEW

**Comment by daresheep** [comment]
CLA signature

File: pkg/hostpath/controllerserver.go [hunk 1]
```diff
@@ -413,20 +413,26 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
...
+	//since other process need check current snapshot id
 	snapshot := hostPathSnapshot{}
 	snapshot.Name = req.GetName()
 	snapshot.Id = snapshotID
 	snapshot.VolID = volumeID
 	snapshot.Path = file
 	snapshot.CreationTime = creationTime
 	snapshot.SizeBytes = hostPathVolume.VolSize
+	snapshot.ReadyToUse = false
+	hostPathVolumeSnapshots[snapshotID] = snapshot
```
Comment (aayushrangwala) [inline pkg/hostpath/controllerserver.go:426]: This entry is being added below as well `:438` which makes more sense there as anyway the snapshot_id is being added to map only after the successful creation.

File: pkg/hostpath/controllerserver.go [hunk 2]
```diff
@@ -413,20 +413,26 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
...
-	glog.V(4).Infof("create volume snapshot %s", file)
+	//since other process need check current snapshot id
 	snapshot := hostPathSnapshot{}
 	snapshot.Name = req.GetName()
 	snapshot.Id = snapshotID
 	snapshot.VolID = volumeID
 	snapshot.Path = file
 	snapshot.CreationTime = creationTime
 	snapshot.SizeBytes = hostPathVolume.VolSize
+	snapshot.ReadyToUse = false
```
Comment (aayushrangwala) [inline pkg/hostpath/controllerserver.go:425]: `snapshot.ReadyToUse` must be false by default as its a bool with zero value `false`. There seems to be no need here

File: pkg/hostpath/controllerserver.go [hunk 3]
```diff
@@ -413,20 +413,26 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
...
+	hostPathVolumeSnapshots[snapshotID] = snapshot
+
+	executor := utilexec.New()
+	out, err := executor.Command(cmd[0], cmd[1:]...).CombinedOutput()
+	if err != nil {
+		delete(hostPathVolumeSnapshots,snapshotID)
+		return nil, status.Errorf(codes.Internal, "failed create snapshot: %v: %s", err, out)
+	}
+
+	glog.V(4).Infof("create volume snapshot %s", file)
```
Comment (aayushrangwala) [inline pkg/hostpath/controllerserver.go:435]: log is incorrect. It should be `Craeted`

**Comment by aayushrangwala** [comment]
This overall fix in the PR doesnot seems to make much difference. The issue seems to be there for the large PVs and its entry is not being added to the map and in the next iteration `getSnapshotByName()` may not be returning anything.

Need to check if the command is executed without error or is it going timeout

**Pull request closed without merging**